core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/BinaryRangeAggregator.java

```diff
@@ -214,6 +223,99 @@ private int collect(int doc, long ord, long bucket, int lowBound) throws IOExcep
         protected abstract void doCollect(LeafBucketCollector sub, int doc, long bucket) throws IOException;
     }
 
+    static abstract class SortedBinaryRangeLeafCollector extends LeafBucketCollectorBase {
```

Comment:
I thought about it but could not find a way to do it cleanly without relying on boxed Longs, which would make things slower.